File: tests/samples/sample.ts
function nested_function(items: number[], lookup: number[]): number {
  let total = 0;
  for (const item of items) {
    for (const other of lookup) {
      if (item === other) {
        total += 1;
      }
    }
  }
  return total;
}

function allocation_function(items: number[]): number[] {
  const values: number[] = [];
  for (const item of items) {
    values.push(item);
  }
  return values;
}

function clean_function(value: number): number {
  return value * 2;
}

function recursive_function(n: number): number {
  if (n <= 1) {
    return n;
  }
  return recursive_function(n - 1) + recursive_function(n - 2);
}

function concat_function(parts: string[]): string {
  let text = "";
  for (const part of parts) {
    text = text + part;
  }
  return text;
}

File: tests/samples/sample.cpp
int nested_function(int items[], int lookup[], int size) {
    int total = 0;
    for (int i = 0; i < size; i++) {
        for (int j = 0; j < size; j++) {
            if (items[i] == lookup[j]) {
                total += 1;
            }
        }
    }
    return total;
}

int allocation_function(int size) {
    int total = 0;
    for (int i = 0; i < size; i++) {
        auto value = new int(i);
        total += *value;
    }
    return total;
}

int clean_function(int value) {
    return value * 2;
}

int recursive_function(int n) {
    if (n <= 1) {
        return n;
    }
    return recursive_function(n - 1) + recursive_function(n - 2);
}

std::string concat_function(std::vector<std::string> parts) {
    std::string text = "";
    for (auto part : parts) {
        text = text + part;
    }
    return text;
}
